crane/unit_of_work.go

```diff
@@ -289,3 +293,55 @@ func (uow *UnitOfWork) ensureInRequireStarted(name string) {
 		uow.requireStarted = append(uow.requireStarted, name)
 	}
 }
+
+func (uow *UnitOfWork) prepareRequirements() {
+	uow.prepareNetworks()
+	uow.prepareVolumes()
+}
+
+func (uow *UnitOfWork) requiredNetworks() []Network {
+	required := []Network{}
+	networks := cfg.NetworkNames()
+	if len(networks) == 0 {
+		return required
+	}
+	for _, container := range uow.Containers() {
+		net := container.RunParams().Net()
+		if includes(networks, net) {
+			required = append(required, cfg.Network(net))
+		}
+	}
+	return required
+}
+
+func (uow *UnitOfWork) requiredVolumes() []Volume {
+	required := []Volume{}
+	volumes := cfg.VolumeNames()
+	if len(volumes) == 0 {
+		return required
+	}
+	for _, container := range uow.Containers() {
+		for _, volumeSource := range container.RunParams().VolumeSources() {
+			if includes(volumes, volumeSource) {
+				required = append(required, cfg.Volume(volumeSource))
+			}
+		}
+	}
+	return required
+}
+
+func (uow *UnitOfWork) prepareNetworks() {
+	for _, n := range uow.requiredNetworks() {
```

Comment:
Maybe these "required" slices should be exposed so that the debug "Command will be applied to" can reference them as well?